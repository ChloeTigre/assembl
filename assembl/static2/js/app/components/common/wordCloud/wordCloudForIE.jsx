// @flow
import React, { Fragment } from 'react';
// Type imports
import type { Keyword, Keywords } from '../../../pages/semanticAnalysis/dataType';

export type Props = {
  /** Array of Keywords */
  keywords: Keywords,
  /** Optional maximum number of keywords to show */
  numberOfKeywordsToDisplay: number,
  /** Optional function called when a word is clicked */
  onKeywordClick: (word: Keyword) => void
};

const DEFAULT_NUMBER_OF_KEYWORDS_TO_DISPLAY = 20;
const KEYWORD_REM = 30;

const style = {
  cursor: 'pointer'
};

/**
 * Function that generates the font size for a keyword
 *
 * @param {Int} score keyword score use to compute the font weight from a base KEYWORD_REM
 */
const computeFontSize = (score: number) => KEYWORD_REM * score;

/**
 * Function that generates a keyword embedded in an anchor in a paragraph
 *
 * @param {Keyword} k Current keyword
 * @param {Int} i Current keyword index used as key to avoid duplicate React entries
 */
const keyword = (k: Keyword, i: number, onKeywordClick: (word: Keyword) => void) => (
  <div key={i} style={{ ...style, fontSize: `${computeFontSize(k.score)}px` }} onClick={() => onKeywordClick(k)}>
    <a>{k.value}</a>
  </div>
);

const WordCloudForIE = ({ keywords, numberOfKeywordsToDisplay, onKeywordClick }: Props) => (
  <Fragment>{keywords.slice(0, numberOfKeywordsToDisplay).map((k, i) => keyword(k, i, onKeywordClick))}</Fragment>
);

WordCloudForIE.defaultProps = {
  numberOfKeywordsToDisplay: DEFAULT_NUMBER_OF_KEYWORDS_TO_DISPLAY,
  onKeywordClick: () => {}
};

export default WordCloudForIE;