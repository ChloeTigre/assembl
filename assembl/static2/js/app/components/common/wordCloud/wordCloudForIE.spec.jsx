// @flow
import React from 'react';
/* eslint-disable import/no-extraneous-dependencies */
import initStoryshots from '@storybook/addon-storyshots';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
/* eslint-enable */

import WordCloudForIE from './wordCloudForIE';
import { defaultWordCloudForIEProps } from './wordCloudForIE.stories';

initStoryshots({
  storyKindRegex: /^Semantic\s{1}Analysis\|WordCloudForIE$/
});

configure({ adapter: new Adapter() });

describe('<WordCloudForIE /> - with shallow', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<WordCloudForIE {...defaultWordCloudForIEProps} />);
  });

  it('should render keywords in div', () => {
    expect(wrapper.find('div')).toHaveLength(2);
  });

  it('should not render keywords', () => {
    const props = { keywords: [] };
    wrapper = shallow(<WordCloudForIE {...props} />);
  });
});