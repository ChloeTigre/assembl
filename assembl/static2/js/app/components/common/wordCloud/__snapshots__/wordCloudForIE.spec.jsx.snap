// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Storyshots Semantic Analysis|WordCloudForIE default 1`] = `
Array [
  <div
    onClick={[Function]}
    style={
      Object {
        "cursor": "pointer",
        "fontSize": "9px",
      }
    }
  >
    <a>
      text five
    </a>
  </div>,
  <div
    onClick={[Function]}
    style={
      Object {
        "cursor": "pointer",
        "fontSize": "6px",
      }
    }
  >
    <a>
      text three
    </a>
  </div>,
]
`;

exports[`Storyshots Semantic Analysis|WordCloudForIE playground 1`] = `
Array [
  <div
    onClick={[Function]}
    style={
      Object {
        "cursor": "pointer",
        "fontSize": "9px",
      }
    }
  >
    <a>
      text five
    </a>
  </div>,
  <div
    onClick={[Function]}
    style={
      Object {
        "cursor": "pointer",
        "fontSize": "6px",
      }
    }
  >
    <a>
      text three
    </a>
  </div>,
]
`;
